web/packages/teleport/src/Roles/RoleEditor/StandardEditor/MetadataSection.tsx

```diff
@@ -16,70 +16,156 @@
-import { precomputed } from 'shared/components/Validation/rules';
+import { FieldTextArea } from 'shared/components/FieldTextArea';
+import { precomputed, requiredAll } from 'shared/components/Validation/rules';
+import { useAsync } from 'shared/hooks/useAsync';
+import { debounce } from 'shared/utils/highbar';
 
 import { LabelsInput } from 'teleport/components/LabelsInput';
+import { ApiError } from 'teleport/services/api/parseError';
+import useTeleport from 'teleport/useTeleport';
 
-import { SectionBox, SectionPadding, SectionProps } from './sections';
+import { SectionPadding, SectionPropsWithDispatch } from './sections';
 import { MetadataModel, roleVersionOptions } from './standardmodel';
+import { ActionType } from './useStandardModel';
 import { MetadataValidationResult } from './validation';
 
-export const MetadataSection = ({
-  value,
-  isProcessing,
-  validation,
-  onChange,
-}: SectionProps<MetadataModel, MetadataValidationResult>) => (
-  <Flex flexDirection="column" gap={3}>
-    <SectionPadding>Basic information about this role</SectionPadding>
-    <SectionBox
-      titleSegments={['Role Information']}
-      isProcessing={isProcessing}
-      validation={validation}
-    >
-      <FieldInput
-        label="Role Name"
-        required
-        placeholder="Enter Role Name"
-        value={value.name}
-        disabled={isProcessing}
-        rule={precomputed(validation.fields.name)}
-        onChange={e => onChange({ ...value, name: e.target.value })}
-      />
-      <FieldInput
-        label="Description"
-        placeholder="Enter Role Description"
-        value={value.description || ''}
-        disabled={isProcessing}
-        onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
-          onChange({ ...value, description: e.target.value })
-        }
-      />
-      <Box mb={3}>
-        <Text typography="body3" mb={1}>
-          Labels
-        </Text>
-        <LabelsInput
-          disableBtns={isProcessing}
-          labels={value.labels}
-          setLabels={labels => onChange({ ...value, labels })}
-          rule={precomputed(validation.fields.labels)}
-        />
-      </Box>
-      <FieldSelect
-        label="Version"
-        isDisabled={isProcessing}
-        options={roleVersionOptions}
-        value={value.version}
-        onChange={version => onChange({ ...value, version })}
-        mb={0}
-        menuPosition="fixed"
-      />
-    </SectionBox>
-  </Flex>
+export const MetadataSection = memo(
+  ({
+    value,
+    isProcessing,
+    validation,
+    isEditing,
+    dispatch,
+  }: SectionPropsWithDispatch<MetadataModel, MetadataValidationResult> & {
+    isEditing: boolean;
+  }) => {
+    const theme = useTheme();
+    const { resourceService } = useTeleport();
+
+    // Verifies whether a role already exists with this name. This will be used
+    // to show a validation error if necessary.
+    const [, checkForNameCollisionNow] = useAsync(
```

Comment:
Another approach wrt naming would be to use the regular form for non-debounced function `checkForNameCollision` and use `Debounced` suffix for the debounced version. Though maybe with your version it's easier to avoid calling the non-debounced version by mistake.